proto/spire/api/node/node.proto

```diff
@@ -32,8 +32,9 @@ message X509SVID {
 // A message returned by the Spire Server, which includes a map of signed SVIDs and
 //a list of all current Registration Entries which are relevant to the caller SPIFFE ID.
 message X509SVIDUpdate {
-    // A map containing SVID values and corresponding SPIFFE IDs as the
-    // keys. Map[SPIFFE_ID] => SVID.
+    // A map containing SVID values keyed by:
+    //  - SPIFFE ID in message 'AttestResponse'        (Map[SPIFFE_ID] => SVID)
+    //  - Entry  ID in message 'FetchX509SVIDResponse' (Map[Entry_ID]  => SVID)
     map<string, X509SVID> svids = 1;
```

Comment:
Sorry for the late comment on this bit, but I was discussing this change with @azdagron and we wondered if we should also deprecate this field since the key is changing and clients may be dependent on the behavior of the key. Not quite sure how difficult it would be to do that... what do you think?

Reply:
No problem! It is a simple change to do, but I'm not sure if we should since the underlying data structure remains the same. Currently, the handler figures out the key to use in the response based on the agent request.

There is also a drawback, the `Attest()` RPC would have to return two SVIDs maps holding the same information. Unlike the `FetchX509SVID()`, there is not a way to identify if the request comes from a legacy agent or not without modifying the `AttestRequest` message.

I'll push it so we can better see how it would be and suggest improvements. I can revert it later if we decide to keep it as it is now.

Reply:
> Currently, the handler figures out the key to use in the response based on the agent request.
>
> There is also a drawback, the Attest() RPC would have to return two SVIDs maps holding the same information. Unlike the FetchX509SVID(), there is not a way to identify if the request comes from a legacy agent or not without modifying the AttestRequest message.

Hmm ok yes those are good points... And now that I see the change, it does feel unnecessary. Sorry to derail you here. This PR looks good to me!

Reply:
no worries! :slightly_smiling_face: I'll revert it